docs/source/guide/tasks.md

```diff
@@ -102,6 +103,116 @@ http://example.com/text.txt
+
+#### Variables 
+
+In most cases, the `Object` tag has the value with one variable (prefixed with a $) in it. For example, `<Audio value="$audio" ... />`. All variables  point to field in task data. For example, `$audio => task.data.audio`
+
+!!! attention "important"
+    In most cases, the `Object` tag has the value with one variable (prefixed with a $) in it. For example, `<Audio value="$audio" ... />`. All the variables point to a field in task data. For example, `$audio => task.data.audio`.
+
+
+#### Text
+
+The value parameter can be a string. It is useful for `Header` and `Text`. 
+
+Also, you can use the content of the tag as value. It is useful for descriptive text tags and is applied for `Label` and `Choice`.
+
+For example:
+
+```html
+<Header>Label audio:</Header>
+<Header value="Label only fully visible cars" />
+<Text name="instruction" value="Label only fully visible cars" />
+<Label>cat</Label>
+
+<Choice>other</Choice>
+```
+
+#### Other cases
+
+1. The `value` parameter can be a text containing variables prefixed by $.
+
+    For example:
+    ```html
+    <Header value="url: $image"/>
+    ```
+
+2. The `value` parameter can also refer to nested data in arrays and dicts (`$texts[2]` and `$audio.url`). 
+
+    For example: 
+    ```html
+    <Image name="image" value="$images[0]"/>
+    ```
+
+3. The `value` parameter can include `Repeater` tag substitution, by default `{{idx}} ($audios[{{idx}}].url)`. 
+The `value` parameter can be a string. It is useful for `Header` and `Text`. 
+
+
+### `valueType` (optional)
+
+The `valueType` parameter defines how to treat the data retrieved from the previous steps.
+There are two options such as the  "url" and raw data. Currently the raw data input can be  "text” or "json”. The  “text” is used for `HyperText` and `Text` tags and "json" is used for `TimeSeries` tag. 
+
+For example:
+
+- Using “url”: `<Text name="text1" value="$text" valueType="url"/>` displays the text loaded by the URL.
+
+- Using “text”: `<Text name="text" value="$text" valueType="text"/>` displays the URL without loading the text.
+
+### `resolver` (optional)
+
+Task data can point to a data file on S3 storage or cloud storage with the actual data. You can retrieve it only in run-time.
+
+To retrieve the task data, use rules from the `resolver` parameter. 
+
+If you import a file with a list of tasks, and every task in this list is a link to another file in the storage. In this case, you can use the `resolver` parameter to retrieve the content of these files from a storage. 
+
+**Use case**
+
+When the user has a CSV file with a list of tasks. The column “S3” in the CSV file is a link to another CSV file in the storage. Every CSV file has a “url” column. The “url” column is a link to the text file which is retrieved for labeling.
+
+**Solution**
+
+To retrieve the file, use the following parameters:
+
+1. `value="S3"`: The URL to CSV on S3. If you use the `resolver` parameter the value is treated as a URL.
```

Comment:
I don't understand this option. Does "S3" mean the link or what?

Reply:
rephrased